### ISSUE TITLE
Game freezes when opening menus after updating from v21 to v22.5

### ISSUE THREAD

**Issue by Dumlupe** [llde/xOBSE#132]
I have updated my game from v21 (the pre "x" version) to v22.5. I get the message that it's updated successfully and everything seems to be working fine. However when I open a menu (both the escape menu and the tab menu) my game freezes and nothing I do seems to make it unfreeze. This only affects existing save games, a new save game seems to work ok, but I'd rather not have to do that if possible.

I am using the following OBSE plugins:
![ce04c0bf2d2ed1c976c813dd7309dbfb (1)](https://user-images.githubusercontent.com/98250248/150667128-f91305db-9ffa-45e6-8feb-b4f38aaccf73.png)

And the following obse-using mods (I believe this is all of them but I'm not 100% sure):

Maskar's Oblivion Overhaul
Maskar's Unarmored Skill
Ultimate Leveling
LINK
FEA - Fundamental Enchanting Addons
Av Latta Magicka
Fundament

I am also using ENBoost and a 4gb patch although I doubt those are causing the issue.

**Comment by llde** [comment]
I would like 2 thing:
1)Upload the save file and the OBSE cosave
2) Try to test by progressivley disabling mods if the problem disappear, and provide the name of the mod

Thanks for the report

**Comment by Dumlupe** [comment]
Here is the save I first encountered the bug:
[Save.zip](https://github.com/llde/xOBSE/files/7924529/Save.zip)

As for which mod is causing it, I unfortunately wasn't able to figure it out. I tried progressively disabling all the obse plugins, one by one, then disabling all of my mods, one by one, and neither seemed to fix anything. However, there were two mods which I was unable to test because the save wouldn't load without them: Maskar's Oblivion Overhaul and a mod I initially missed called class advantages:
[https://www.nexusmods.com/oblivion/mods/42780](url)
 [https://www.nexusmods.com/oblivion/mods/35300](url)

It's possible one of these two is at fault, or it could be there's something I overlooked, like a specific combination of mods or plugins. I will continue testing and let you know if I find anything.

**Comment by llde** [comment]
Thanks I hope it's enough info

**Comment by llde** [comment]
@Dumlupe  Hi. 
The save you provided seems to crash consistently on load.
Can you provide a save that already stripped from not needed mods (keep just the basic one, UOP for example, and the 2 mods required to not make it hang)?

**Comment by Dumlupe** [comment]
Yes, here is a save with absolutely everything removed except the dlcs, the unofficial patches, and the two mods.  It had to be a quicksave because opening the save menu still freezes the game with all but these two mods removed - which makes me think one of them must be at fault somehow.

[Save 2.zip](https://github.com/llde/xOBSE/files/7968775/Save.2.zip)

**Comment by llde** [comment]
It's seems to be hanging during the Cleanup after a Container Item Loop.
Thisa may be the result of having a mod that used the Inventory References with pre xOBSE 22, that was problematic and was leaving waste in  some case.
I'm gonna see tomorrow if it's possible to make this more robust.

**Comment by llde** [comment]
@Dumlupe Can you see if it's fixed on your end too using this DLL https://ci.appveyor.com/api/buildjobs/y8k4g2efotml0451/artifacts/obse%2Fobse%2FRelease%201_2_0_416%2Fobse.dll ?

**Comment by Dumlupe** [comment]
Yes, it works! I had to rename the file to have the same numbers at the end as the release version but after that it seemed to work fine. I've enabled all my mods again and everything seems to be working ok. Thank you so much for your help.
